techmatters/hrm
# feat: associate/disassociate profile flags

## Description
This PR adds a new endpoint, `/:profileId/flags/:profileFlagId`, which accepts two methods:
- `POST`, creates a new association between the profile of id `profileId` and the flag of id `profileFlagId` (if it does not exists).
- `DELETE`, deletes the association between the profile of id `profileId` and the flag of id `profileFlagId` (if it exists).

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-2290)

### Verification steps
Trust service tests or
- Start local server.
- Create a new profile. At least one flag will exist when the migration is executed (default flags).
- Hit `/:profileId/flags/:profileFlagId` with a `POST` method, where `profileId` value is the above profile and `profileFlagId` value is one of the default flags.
- Confirm a new association has been created and that the profile payload contains it under `profileFlags` proerty.
- Hit `/:profileId/flags/:profileFlagId` with a `DELETE` method, where `profileId` value is the above profile and `profileFlagId` value is one of the default flags.
- Confirm that the association has been removed and that the profile payload does not contains the flag anymore.
